wfe/web-front-end.go

```diff
@@ -298,9 +297,22 @@ const (
 	malformedJWS = "Unable to read/verify body"
 )
 
+// verifyPOST reads and parses the request body, looks up the Registration
+// corresponding to its JWK, verifies the JWS signature,
+// checks that the resource field is present and correct in the JWS protected
+// header, and returns the JWS payload bytes, the key used to verify, and the
+// corresponding Registration (or error).
+// If regCheck is false, verifyPOST will still try to look up a registration
+// object, and will return it if found. However, if no registration object is
+// found, verifyPOST will attempt to verify the JWS using the key in the JWS
+// headers, and return the key plus a dummy registration if successful. If a
+// caller passes regCheck = false, it should plan on validating the key itself.
 func (wfe *WebFrontEndImpl) verifyPOST(request *http.Request, regCheck bool, resource core.AcmeResource) ([]byte, *jose.JsonWebKey, core.Registration, error) {
```

Comment:
The verifyPost function is now 120 lines long. It's gotten complicated enough that I think we need to step back and consider a refactor.

Could we maybe have some kind of RegRequired filter pulls the Registration for all of the functions that need it and then passes it to the next Handler in the chain? (Not a http.Handler but some kind of RegistrationHandler that requires a Registration to be passed to it.) The Required one would make sure the JWK given by the user matched on in the database. RegDoesntExistHandler or something either pull it from the header. We could use that Registration to do the rest of this work.

I know this is might be a PITA because we allowed multiple verbs on the same path and they may have different registration expectations, but we can fix that up with `github.com/bmizerany/pat`.

Also, I note that this file is over 1200 lines and that's pretty big for a Go file. Breaking it up some might help things.